cdm/core/src/main/java/ucar/nc2/Group.java

```diff
@@ -374,20 +376,56 @@ public boolean removeAttributeIgnoreCase(String attName) {
 
   ////////////////////////////////////////////////////////////////////////
 
-  /** Find a Enumeration in this or a parent Group, using its short name. */
+  /** Find a Enumeration in this Group, using its short name. */
   @Nullable
   public EnumTypedef findEnumeration(String name) {
+    return findEnumeration(name, false); // Keep the old behavior
+  }
+
+  /** Find a Enumeration in this or optionally the parent Groups, using its short name. */
+  @Nullable
+  public EnumTypedef findEnumeration(String name, boolean searchup) {
     if (name == null)
       return null;
     // name = NetcdfFile.makeNameUnescaped(name);
-    for (EnumTypedef d : enumTypedefs) {
+    // search this group's EnumTypedefs
+    for (EnumTypedef d : this.getEnumTypedefs()) {
       if (name.equals(d.getShortName()))
         return d;
     }
-    Group parent = getParentGroup();
-    if (parent != null)
-      return parent.findEnumeration(name);
+    if (searchup) {
+      Group parent = getParentGroup();
+      if (parent != null)
+        return parent.findEnumeration(name, searchup);
+    }
+    return null;
+  }
 
+  /**
+   * Locate an enum type definition that is structurally
+   * similar to the template type def. The Enum names are ignored.
+   *
+   * @param template match this enum type def
+   * @param searchup if true, then search this group and then parent groups.
+   */
+  public EnumTypedef findSimilarEnumTypedef(EnumTypedef template, boolean searchup) {
+    EnumTypedef ed = null;
+    assert (template != null);
+    // search this group builders's EnumTypedefs but with constraint on name
+    {
+      Optional<EnumTypedef> edopt = this.enumTypedefs.stream().filter(e -> (e.equalsMapOnly(template))).findFirst();
+      ed = (edopt.isPresent() ? edopt.get() : null);
+    }
```

Comment:
I think this can be simplified to
```suggestion
    assert (template != null);
    // search this group builders's EnumTypedefs but with constraint on name
    EnumTypedef ed = this.enumTypedefs.stream().filter(e -> e.equalsMapOnly(template)).findFirst().orElse(null);
```

Reply:
done